examples/fewshot_protonet/README.md

```diff
@@ -0,0 +1,102 @@
+# Prototypical Networks for Few-shot Learning
+
+### 1. Description
+
+This demo implements [Prototypical Networks for Few-shot Learning](https://github.com/jakesnell/prototypical-networks) within the `PyKale` framework. The following instructions are about how to use it.
+
+ProtoNet is for few-shot learning under $N$-Way $K$-Shot settings:
+
+$N$-way: It means the number of classes or categories in one epoch in evaluation. For instance, in a 5-way scenario, data from five classes is fed into the model in an epoch.
+
+$K$-shot: This is the number of examples (referred to as "shots") from each class in training and evaluation. A 1-shot learning task indicates only one example per class, while a 3-shot task has three examples per class.
+
+## 2. Usage
+
+### Datasets
+
+This data loader can be used on several few-shot learning datasets, such as mini-ImageNet ([official image list](https://drive.google.com/file/d/1iBu_Iqt49opXHSUNcTRU2WQas1WICLwQ/view)/[ready-to-download data on Kaggle](https://www.kaggle.com/datasets/arjunashok33/miniimagenet)), tiered-ImageNet ([official dataset-generating tool](https://github.com/yaoyao-liu/tiered-imagenet-tools)/[ready-to-download data on Kaggle](https://www.kaggle.com/datasets/arjun2000ashok/tieredimagenet)), and Omniglot ([official downloading code](https://github.com/brendenlake/omniglot)/[ready-to-download data on Kaggle](https://www.kaggle.com/datasets/watesoyan/omniglot)), etc. All datasets should be organized as follows.
+
+<pre>
+└── root
+    ├── train
+    |   ├── class 1
+    |   |   ├── image 1
+    |   |   ├── image 2
+    |   |   └── ...
+    |   ├── class 2
+    |   |   ├── image 1
+    |   |   ├── image 2
+    |   |   └── ...
+    |   └── ...
+    ├── val
+    |   ├── class m
+    |   |   ├── image 1
+    |   |   ├── image 2
+    |   |   └──...
+    |   ├── class m+1
+    |   |   ├── image 1
+    |   |   ├── image 2
+    |   |   └──...
+    |   └── ...
+    └── test (optional)
+        ├── class n
+        |   ├── image 1
+        |   ├── image 2
+        |   └── ...
+        ├── class n+1
+        |   ├── image 1
+        |   ├── image 2
+        |   └── ...
+        └── ...
+</pre>
+
+### Examples
+
+#### Single GPU training:
+
+Example - Training ResNet18 on Omniglot in a 5-way-5-shot protocol
+
+`python main.py --cfg configs/demo.yaml --gpus 1`
+
+Customized running
+
+`python main.py --cfg configs/template.yaml --gpus 1`
+
+- `demo.yaml` is a demo configuration file. Change `DATASET.ROOT` item in it to fit your dataset directory for a quick demo running.
+- `template.yaml` contains all hyperparameters that users can change. It allows users to customize the model accordingly.
+- Available backbones: any `resnet` structures from `torchvision.models` or `kale.embed.image_cnn`.
+
+
+#### Test
+
+Example - Testing ResNet18 on Omniglot in a 5-way-5-shot protocol
```

Comment:
"protocal"? See above. Check and fix all. I will not repeatedly comment on this.